File: src (cpp)/infrastructure/cell.cpp
#include "cell.h"

Cell::Cell(bool accessible) {
  is_accessible = accessible;
}
Cell::Cell(Cell& C) {
  is_accessible = C.is_accessible;
}
Cell& Cell::operator=(Cell& C) {
  is_accessible = C.is_accessible;
  return *this;
}
bool Cell::IsAccessible() const {
  return is_accessible;
}

File: src/virtualzoo/infrastructure/Cell.java
package virtualzoo.infrastructure;

import virtualzoo.misc.Renderable;

/**
 * Kelas Cell yang mendefinisikan aksesibilitas suatu cell.
 *
 * @author Felix Limanta - 13515065
 * @version 3.0
 * @since 3.0
 */
public abstract class Cell implements Renderable {

  /**
   * Suatu Cell dapat dikunjungi atau tidak oleh pengunjung
   */
  protected boolean infrastructureAccessible;

  /**
   * Constructor
   * Menciptakan cell default yang dapat/tidak dapat diakses.
   *
   * @param accessible Nilai true/false cell dapat diakses.
   */
  public Cell(boolean accessible) {
    infrastructureAccessible = accessible;
  }

  /**
   * Getter nilai infrastructureAccessible
   *
   * @return Nilai infrastructureAccessible
   */
  public boolean isAccessible() {
    return infrastructureAccessible;
  }
}
